src/operations.jl

```diff
@@ -1,18 +1,23 @@
-struct TransformsRow{T, F}
+struct TransformsRow{T, F} <: AbstractRow
     row::T
     funcs::F
 end
 
-Base.getproperty(row::TransformsRow, nm::Symbol) = (getfunc(row, getfield(row, 2), nm))(getproperty(getfield(row, 1), nm))
-Base.propertynames(row::TransformsRow) = propertynames(getfield(row, 1))
+getrow(r::TransformsRow) = getfield(r, :row)
+getfuncs(r::TransformsRow) = getfield(r, :funcs)
 
-struct Transforms{C, T, F}
+getcolumn(row::TransformsRow, nm::Symbol) = (getfunc(row, getfuncs(row), nm))(getcolumn(getrow(row), nm))
+getcolumn(row::TransformsRow, i::Int) = (getfunc(row, getfuncs(row), i))(getcolumn(getrow(row), i))
+columnnames(row::TransformsRow) = columnnames(getrow(row))
+
+struct Transforms{C, T, F} <: AbstractColumns
     source::T
     funcs::F # NamedTuple of columnname=>transform function
 end
 
-Base.propertynames(t::Transforms{true}) = propertynames(getfield(t, 1))
-Base.getproperty(t::Transforms{true}, nm::Symbol) = Base.map(getfunc(t, getfield(t, 2), nm), getproperty(getfield(t, 1), nm))
+columnnames(t::Transforms{true}) = columnnames(getfield(t, 1))
+getcolumn(t::Transforms{true}, nm::Symbol) = Base.map(getfunc(t, getfield(t, 2), nm), getcolumn(getfield(t, 1), nm))
+getcolumn(t::Transforms{true}, i::Int) = Base.map(getfunc(t, getfield(t, 2), nm), getcolumn(getfield(t, 1), i))
```

Comment:
```suggestion
getcolumn(t::Transforms{true}, i::Int) = Base.map(getfunc(t, getfield(t, 2), i), getcolumn(getfield(t, 1), i))
```
(Probably needs a test.)